### PR TITLE
Merge master into post-dev15

### PR DESCRIPTION
**Pull request by vslsnap**
This is an automatically generated pull request from master into post-dev15.

@dotnet/roslyn-infrastructure:

``` bash
git remote add vslsnap-roslyn "https://github.com/vslsnap/roslyn.git"
git fetch vslsnap-roslyn
git fetch upstream
git checkout merge-master-into-post-dev15-20170206-100158
git reset --hard upstream/post-dev15
git merge upstream/master
# Fix merge conflicts
git commit
git push vslsnap-roslyn merge-master-into-post-dev15-20170206-100158 --force
```

Once all conflicts are resolved and all the tests pass, you are free to merge the pull request.

### CODE REVIEW

**Comment by vslsnap** [comment]
@dotnet-bot test vsi please

**Comment by jmarolf** [comment]
test windows_release_unit64_prtest  please

**Comment by jmarolf** [comment]
@heejaechang can you investigate these test failures and see If we've  made a bad merge here?

File: src/Workspaces/Remote/ServiceHub/Shared/ServiceHubServiceBase.cs [hunk 1]
```diff
@@ -48,6 +49,26 @@ protected ServiceHubServiceBase(Stream stream, IServiceProvider serviceProvider)
...
+            Logger.TraceInformation($"{DebugInstanceString} Service instance created");
+
+            _cancellationTokenSource = new CancellationTokenSource();
+            CancellationToken = _cancellationTokenSource.Token;
+
+            // due to this issue - https://github.com/dotnet/roslyn/issues/16900#issuecomment-277378950
+            // all sub type must explicitly start JsonRpc once everything is
+            // setup
+            Rpc = new JsonRpc(stream, stream, this);
+            Rpc.Disconnected += OnRpcDisconnected;
```
Comment (heejaechang) [inline src/Workspaces/Remote/ServiceHub/Shared/ServiceHubServiceBase.cs]: this is missing -  Rpc.JsonSerializer.Converters.Add(AggregateJsonConverter.Instance);

**Pull request merged**